lib/screens/package_details/package_details_bloc.dart

```diff
@@ -0,0 +1,56 @@
+import 'dart:async';
+
+import 'package:bloc/bloc.dart';
+import 'package:flutter/foundation.dart';
+import 'package:pub_client/pub_client.dart';
+import 'package:tavern/screens/bloc.dart';
+import 'package:tavern/screens/package_details/package_details_event.dart';
+import 'package:tavern/src/cache.dart';
+
+class PackageDetailsBloc
+    extends Bloc<PackageDetailsEvent, PackageDetailsState> {
+  final PubHtmlParsingClient client;
+  final PackageRepository _packageRepository;
+
+  PackageDetailsBloc({@required this.client})
+      : _packageRepository = PackageRepository(client: client);
+
+  @override
+  PackageDetailsState get initialState => InitialPackageDetailsState();
+
+  @override
+  Stream<PackageDetailsState> mapEventToState(
+    PackageDetailsEvent event,
+  ) async* {
+    if (event is GetPackageDetailsEvent) {
+      FullPackage package = await _packageRepository.get(event.packageName);
+
+      yield PackageDetailsState(package: package);
+      event.onComplete.complete();
```

Comment:
why is this necessary?

Reply:
To notify the widget that's waiting for the completed future. I'm making my own future since it's required